### ISSUE TITLE
run

### ISSUE THREAD

**Issue by lengmianlaotang** [haooozi/P2P#2]
Dear Author,

I apologize for bothering you again. I ran the P2P code, but the loss became negative, which has left me a bit confused. My run information is as follows. Thank you again for your response.

python train.py --config configs/voxel/kitti/car.py
09/18 15:29:03 - mmengine - WARNING - The prefix is not set in metric class TrackAccuracy.
09/18 15:29:04 - mmengine - INFO - 
------------------------------------------------------------
System environment:
    sys.platform: linux
    Python: 3.9.19 | packaged by conda-forge | (main, Mar 20 2024, 12:50:21) [GCC 12.3.0]
    CUDA available: True
    numpy_random_seed: 277622604
    GPU 0: NVIDIA GeForce RTX 3090
    CUDA_HOME: /usr/local/cuda-11.1
    NVCC: Cuda compilation tools, release 11.1, V11.1.74
    GCC: gcc (Ubuntu 9.4.0-1ubuntu1~20.04.2) 9.4.0
    PyTorch: 2.0.1+cu117
    PyTorch compiling details: PyTorch built with:
  - GCC 9.3
  - C++ Version: 201703
  - Intel(R) oneAPI Math Kernel Library Version 2022.2-Product Build 20220804 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v2.7.3 (Git Hash 6dbeffbae1f23cbbeae17adb7b5b13f1f37c080e)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - LAPACK is enabled (usually provided by MKL)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 11.7
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86
  - CuDNN 8.5
  - Magma 2.6.1
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=11.7, CUDNN_VERSION=8.5.0, CXX_COMPILER=/opt/rh/devtoolset-9/root/usr/bin/c++, CXX_FLAGS= -D_GLIBCXX_USE_CXX11_ABI=0 -fabi-version=11 -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -DNDEBUG -DUSE_KINETO -DLIBKINETO_NOROCTRACER -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -O2 -fPIC -Wall -Wextra -Werror=return-type -Werror=non-virtual-dtor -Werror=bool-operation -Wnarrowing -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wunused-local-typedefs -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Werror=cast-function-type -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_DISABLE_GPU_ASSERTS=ON, TORCH_VERSION=2.0.1, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=1, USE_NNPACK=ON, USE_OPENMP=ON, USE_ROCM=OFF, 

    TorchVision: 0.15.2+cu117
    OpenCV: 4.10.0
    MMEngine: 0.7.4

Runtime environment:
    cudnn_benchmark: False
    mp_cfg: {'mp_start_method': 'fork', 'opencv_num_threads': 0}
    dist_cfg: {'backend': 'nccl'}
    seed: 277622604
    Distributed launcher: none
    Distributed training: False
    GPU number: 1

![image](https://github.com/user-attachments/assets/1b9db7f2-ac98-4c8b-ba93-976196ed0ca6)

**Comment by haooozi** [comment]
Hi, no matter for contacting me. We define the loss function in a way that allows for negative values. Therefore, it's not an issue that affects the model. Furthermore, from my knowledge of the SOT task, the loss is not a point of concern, especially for the Siamese paradigm, where the optimal performance may fall on the epoch with high loss.

From: lengmianlaotang ***@***.***>
Date: 2024-09-18 16:03:10
To: haooozi/P2P ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [haooozi/P2P] run (Issue #2)

Dear Author,
I apologize for bothering you again. I ran the P2P code, but the loss became negative, which has left me a bit confused. My run information is as follows. Thank you again for your response.
python train.py --config configs/voxel/kitti/car.py
 09/18 15:29:03 - mmengine - WARNING - The prefix is not set in metric class TrackAccuracy.
 09/18 15:29:04 - mmengine - INFO -System environment:
 sys.platform: linux
 Python: 3.9.19 | packaged by conda-forge | (main, Mar 20 2024, 12:50:21) [GCC 12.3.0]
 CUDA available: True
 numpy_random_seed: 277622604
 GPU 0: NVIDIA GeForce RTX 3090
 CUDA_HOME: /usr/local/cuda-11.1
 NVCC: Cuda compilation tools, release 11.1, V11.1.74
 GCC: gcc (Ubuntu 9.4.0-1ubuntu1~20.04.2) 9.4.0
 PyTorch: 2.0.1+cu117
 PyTorch compiling details: PyTorch built with:
GCC 9.3
C++ Version: 201703
Intel(R) oneAPI Math Kernel Library Version 2022.2-Product Build 20220804 for Intel(R) 64 architecture applications
Intel(R) MKL-DNN v2.7.3 (Git Hash 6dbeffbae1f23cbbeae17adb7b5b13f1f37c080e)
OpenMP 201511 (a.k.a. OpenMP 4.5)
LAPACK is enabled (usually provided by MKL)
NNPACK is enabled
CPU capability usage: AVX2
CUDA Runtime 11.7
NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86
CuDNN 8.5
Magma 2.6.1
Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=11.7, CUDNN_VERSION=8.5.0, CXX_COMPILER=/opt/rh/devtoolset-9/root/usr/bin/c++, CXX_FLAGS= -D_GLIBCXX_USE_CXX11_ABI=0 -fabi-version=11 -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -DNDEBUG -DUSE_KINETO -DLIBKINETO_NOROCTRACER -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -O2 -fPIC -Wall -Wextra -Werror=return-type -Werror=non-virtual-dtor -Werror=bool-operation -Wnarrowing -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wunused-local-typedefs -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Werror=cast-function-type -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_DISABLE_GPU_ASSERTS=ON, TORCH_VERSION=2.0.1, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=1, USE_NNPACK=ON, USE_OPENMP=ON, USE_ROCM=OFF,
TorchVision: 0.15.2+cu117
 OpenCV: 4.10.0
 MMEngine: 0.7.4
Runtime environment:
 cudnn_benchmark: False
 mp_cfg: {'mp_start_method': 'fork', 'opencv_num_threads': 0}
 dist_cfg: {'backend': 'nccl'}
 seed: 277622604
 Distributed launcher: none
 Distributed training: False
 GPU number: 1
image.png (view on web)

**Comment by lengmianlaotang** [comment]
Thank you for your response.

**Comment by lengmianlaotang** [comment]
Dear Author,

Hello. I have reproduced the results based on the configuration file. The results for most of the categories are very close to those reported in the paper, but the results for the "van" category differ significantly. Could you please provide some suggestions? I look forward to your response. Thank you very much.
![image](https://github.com/user-attachments/assets/99302615-1663-47e4-a3c5-9faef09bdc1c)

**Comment by haooozi** [comment]
Maybe you can load the pre-trained car and ped models to train van and cyc, respectively.

**Comment by haooozi** [comment]
For car and pedestrian, fine-grained tuning parameterization (including lr, bs and so on, depending on devices) will repreduct the results, maybe better.

**Comment by haooozi** [comment]
Hello. For car and pedestrian, fine-grained tuning parameterization (including lr, bs and so on, depending on devices) will repreduct the results, maybe better. For van and cyclist, maybe you can load the pre-trained car and ped models to train van and cyc, respectively.

发件人：lengmianlaotang ***@***.***>
发送日期：2024-10-10 15:25:22
收件人：haooozi/P2P ***@***.***>
抄送人：Jiahao Nie ***@***.***>,Comment ***@***.***>
主题：Re: [haooozi/P2P] run (Issue #2)

Dear Author,
Hello. I have reproduced the results based on the configuration file. The results for most of the categories are very close to those reported in the paper, but the results for the "van" category differ significantly. Could you please provide some suggestions? I look forward to your response. Thank you very much.
 image.png (view on web)

**Comment by OFDBTI** [comment]
> 亲爱的作者，
> 
> 很抱歉再次打扰您。我运行了P2P代码，但损失变成了负数，这让我有点困惑。我的运行信息如下。再次感谢您的回复。
> 
> ## python train.py --config configs/voxel/kitti/car.py
> 09/18 15:29:03 - mmengine - 警告 - 度量类 TrackAccuracy 中未设置前缀。09
> /18 15:29:04 - mmengine - 信息 -
> 系统环境： sys.platform：linux Python：3.9.19 | 由 conda-forge 打包 | （main，2024 年 3 月 20 日，12:50:21） [GCC 12.3.0] CUDA 可用：True numpy_random_seed：277622604 GPU 0：NVIDIA GeForce RTX 3090 CUDA_HOME：/usr/local/cuda-11.1 NVCC：Cuda 编译工具，版本 11.1，V11.1.74 GCC：gcc（Ubuntu 9.4.0-1ubuntu1~20.04.2）9.4.0 PyTorch：2.0.1+cu117 PyTorch 编译详细信息： 使用以下工具构建的 PyTorch：
> 
> * GCC 9.3
> * C++ 版本：201703
> * 适用于英特尔(R) 64 位架构应用程序的英特尔(R) oneAPI 数学内核库版本 2022.2 - 产品版本 20220804
> * 英特尔(R) MKL-DNN v2.7.3（Git 哈希 6dbeffbae1f23cbbeae17adb7b5b13f1f37c080e）
> * OpenMP 201511（又名 OpenMP 4.5）
> * LAPACK 已启用（通常由 MKL 提供）
> * NNPACK 已启用
> * CPU 能力使用情况：AVX2
> * CUDA 运行时 11.7
> * NVCC 架构标志：-gencode；arch=compute_37，code=sm_37；-gencode；arch=compute_50，code=sm_50；-gencode；arch=compute_60，code=sm_60；-gencode；arch=compute_70，code=sm_70；-gencode；arch=compute_75，code=sm_75；-gencode；arch=compute_80，code=sm_80；-gencode；arch=compute_86，code=sm_86
> * CuDNN 8.5
> * Magma 2.6.1
> * 构建设置：BLAS_INFO = mkl，BUILD_TYPE = Release，CUDA_VERSION = 11.7，CUDNN_VERSION = 8.5.0，CXX_COMPILER = / opt / rh / devtoolset-9 / root / usr / bin / c ++，CXX_FLAGS = -D_GLIBCXX_USE_CXX11_ABI = 0 -fabi-version = 11 -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -DNDEBUG -DUSE_KINETO -DLIBKINETO_NOROCTRACER -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -O2 -fPIC -Wall -Wextra -Werror = return-type -Werror=非虚拟声明器 -Werror=bool 操作 -Wnarrowing -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wunused-local-typedefs -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Werror=cast-function-type -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_DISABLE_GPU_ASSERTS=ON, TORCH_VERSION=2.0.1, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=1, USE_NNPACK=ON, USE_OPENMP=ON, USE_ROCM=OFF,
>   TorchVision：0.15.2+cu117 
>   OpenCV：4.10.0 
>   MMEngine：0.7.4
> 
> 运行环境： cudnn_benchmark:False mp_cfg:{'mp_start_method':'fork','opencv_num_threads':0} dist_cfg:{'backend':'nccl'} seed:277622604 分布式启动器：无 分布式训练：False GPU数量：1
> 
> ![image](https://github.com/user-attachments/assets/1b9db7f2-ac98-4c8b-ba93-976196ed0ca6)

虽然说作者解释了允许负值，请问您理解了这个损失函数负值的含义了吗？我训练的时候损失刚开始是50，60，随着epoch进行，会逐步降低，经过0之后会继续降低，一般epoch=400时，损失会在-18到-20之间。而在这个区间的损失里，精度和成功率一般最高。大概就是作者说的损失较低时会有最好的表现。
